pkg/apis/environment/common.go

```diff
@@ -93,6 +93,12 @@ func createEnvironment(
 }
 
 func validateEnvironmentCreateInput(r model.EnvironmentCreateInput) error {
+	if err := r.Validate(); err != nil {
+		return err
+	}
+
+	cache := make(map[string]*model.ResourceDefinition)
```

Comment:
No one has been found writing to this cache.

Reply:
@aiwantaozi  it seems that the `dao.GetMatchingResourceDefinitionWith` needs a `cache`.